rancher/rancher
# [2.7] Save OpenLDAP password in Shibboleth config as a secret

## Issue: 

 
## Problem

 
## Solution

 
## Testing


## Engineering Testing
### Manual Testing
Enable the Shibboleth auth provider with OpenLDAP.
Pull the AuthConfig CRD from the cluster and check that the serviceAccountPassword field contains the secret name instead of a plain text credential.
Also verify that the correct password value is stored in the secret that is referenced: kubectl get secrets -n cattle-global-data shibbolethconfig-serviceaccountpassword --template '{{.data.serviceaccountpassword}}' | base64 -d

##Automated Testing
unit tests that verify that the authconfig migration is attempted for unmigrated configs (those that do not have the status block set)

## QA Testing Considerations

 
### Regressions Considerations

It is also worthwhile to check other auth providers since there was a change to how the unstructured object is converted to an AuthConfig.  The updated version is no longer ignoring errors in converting the metadata and then manually converting the metadata chunk.  If you are able to create/save/edit an auth provider, the new code is working correctly.
